### PR TITLE
[RFC] add powerpc64 support

### PR DESCRIPTION
**Pull request by florolf**
(this is a respin of #497 with an added PPC64 target)

Add support for building a pure 64 bit powerpc (big-endian) kernel and userland.

This is based on glibc because musl only has working powerpc64le-support right now, which doesn't seem to have caught on in the embedded word yet.

I've looked at the way other architectures were added and tried to cover the important bits (I can build a firmware for the T4240RDB using this), but I probably have missed something, thus flagging this as RFC.

### CODE REVIEW

File: toolchain/Config.in [hunk 1]
```diff
@@ -238,6 +238,7 @@ comment "C Library"
 choice
 	prompt "C Library implementation" if TOOLCHAINOPTS
 	default LIBC_USE_UCLIBC if arc
+	default LIBC_USE_GLIBC if powerpc64
 	default LIBC_USE_MUSL
```
Comment (blogic) [inline toolchain/Config.in]: i assume musl does not support ppc64 yet ?

**Comment by blogic** [comment]
series looks good. i would like to keep this pending till after the release. i will start a test build today to see how it works.

File: toolchain/Config.in (hunk 1 above)
Comment (florolf) [inline toolchain/Config.in]: It does, but it only implements the v2 ABI. I haven't managed to get a working build of LEDE using that ABI in big-endian mode yet.

Comment (blogic) [inline toolchain/Config.in]: ok, i just started a test build on my server. if it goes well, i'll queue this for post the release

**Comment by nbd168** [comment]
I think you should make a new target instead of adding a subtarget to mpc85xx (which is an entirely different SoC)

**Comment by nbd168** [comment]
ping?

**Comment by florolf** [comment]
Sorry, I tend to miss GitHub notifications. 

I was originally going for making a new target, but then took a look at how other targets handle differing bitness and for example both x86 and malta go for subtargets.

Also, mpc85xx seems to have become a kind of dumping ground for newer PPC implementations (there's a P1020 subtarget).

I have to confess that I still don't really don't understand the level at which LEDE decides whether something constitutes a different target.

So, would you suggest adding a target for ppc64, ex500, or (in my case) t4xxx?

**Comment by nbd168** [comment]
From what I can see, P1020 is basically the same SoC as MPC85xx. I don't think it makes any sense to have a "dumping ground" for PPC implementations.
We typically organize targets by how similar they are in terms of patches, kernel config, etc.
In this case, it probably makes sense to make an ex500 target, since NXP/FreeScale did tend to deviate from other PPC implementations somewhat.

**Comment by commodo** [comment]
@blogic @nbd168 @florolf 
would it be an idea to pull in some of the patches that make sense ?
we can probably defer the `mpc85xx` vs `ex500` part ; 

i'd like to build a T4240 target [ not necessarily have it in the LEDE tree, to not clutter the tree ]
and to do this, some PPC64 bits would be neat ;
the PPC64 bits could make sense to have in the LEDE tree [ if it's ok with you guys ]

**Comment by nbd168** [comment]
Since the Feel free to make a PR for your target which includes relevant

**Comment by nbd168** [comment]
Feel free to make a new PR which includes relevant pieces of this one to make your target work. 
We can then close this one, since there was no update from @florolf

**Comment by commodo** [comment]
i'll keep @florolf as the author of the patches [ if that's ok ]

**Comment by nbd168** [comment]
Sure

**Comment by commodo** [comment]
@nbd168 
sorry for taking too long to get back ; i tried your changes and they work

thanks

**Pull request closed without merging**